init.lua

```diff
@@ -219,18 +218,17 @@ local function can_dig(pos,player)
 	local meta = minetest.get_meta(pos)
 	local inv  = meta:get_inventory()
 	if inv:is_empty('src') and inv:is_empty('dst') then
-		return true
-	else
-		return false
+		return not minetest.is_protected(pos, player)
```

Comment:
Not necessary. Minetest already does this by default. https://github.com/minetest/minetest/blob/master/builtin/game/item.lua#L474